go-controller/pkg/ovn/service.go

```diff
@@ -165,6 +167,130 @@ func (ovn *Controller) syncServices(services []interface{}) {
 	}
 }
 
+func (ovn *Controller) createService(service *kapi.Service) error {
+	klog.V(5).Infof("Creating service %s", service.Name)
+	if !util.IsClusterIPSet(service) {
+		return fmt.Errorf("no cluster IP for service %s found", service.Name)
+	} else if len(service.Spec.Ports) == 0 {
+		return fmt.Errorf("no Ports specified for service")
+	}
+
+	for _, svcPort := range service.Spec.Ports {
+		var port int32
+		if util.ServiceTypeHasNodePort(service) {
+			port = svcPort.NodePort
+		} else {
+			port = svcPort.Port
+		}
+		if port == 0 {
+			continue
```

Comment:
When would port be `0`?

Reply:
It won't be 0 because it'll be rejected by Kube apiserver validation in pkg/apis/core/validation/validation.go::validateServicePort():

```
	for _, msg := range validation.IsValidPortNum(int(sp.Port)) {
		allErrs = append(allErrs, field.Invalid(fldPath.Child("port"), sp.Port, msg))
	}

// IsValidPortNum tests that the argument is a valid, non-zero port number.
func IsValidPortNum(port int) []string {
	if 1 <= port && port <= 65535 {
		return nil
	}
	return []string{InclusiveRangeError(1, 65535)}
}
```

Typically we don't do validation on things that already got validated by the apiserver.

Reply:
I just copied this from the deleteService method. That code was introduced:
https://github.com/ovn-org/ovn-kubernetes/commit/f5282c15d568c19d01ec00664ad1f523c1029f0e#diff-a97e5f1663b5983ab3b31b3c1b630b66R21